plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/util.test.ts

```diff
@@ -85,3 +87,117 @@ describe('isExecutable', () => {
     expect(isExecutable(0o100640)).toBe(false);
   });
 });
+
+describe('parseRepoUrl', () => {
+  const config = new ConfigReader({
+    integrations: {
+      gitlab: [
+        {
+          host: 'www.gitlab.com',
+          token: 'token',
+          apiBaseUrl: 'https://api.gitlab.com',
+        },
+      ],
+      bitbucket: [
+        {
+          host: 'www.bitbucket.net',
+          apiBaseUrl: 'https://api.hosted.bitbucket.net',
+        },
+        {
+          host: 'www.bitbucket.org',
+          username: 'username',
+          appPassword: 'password',
+          apiBaseUrl: 'https://api.hosted.bitbucket.org',
+        },
+      ],
+    },
+  });
+  const integrations = ScmIntegrations.fromConfig(config);
+  it('should throw an exception if no parameters are passed (gitlab)', () => {
+    let e;
+    try {
+      parseRepoUrl('www.gitlab.com', integrations);
```

Comment:
you should be able to completely replace the try/catch logic with something like this

```ts
expect(() => parseRepoUrl('www.gitlab.com', integrations)).toThrow('Invalid repo URL passed to publisher: https://www.gitlab.com/, missing owner')
```